src/utils/async.js

```diff
@@ -0,0 +1,19 @@
+export const timeout = (func, onTimeout = () => {}, timeoutMs = 10000) =>
+  Promise.race([
+    func,
+    typeof timeoutMs === 'number' &&
+      new Promise(resolve => setTimeout(resolve, timeoutMs)).then(onTimeout)
+  ]);
+
+export const tryUntilSuccessful = async (func, maxRetries = 1000, timeoutMs = 1000) => {
+  if (!maxRetries) {
+    return func();
+  }
+
+  try {
+    return await func();
+  } catch (e) {
+    await new Promise(resolve => setTimeout(resolve, timeoutMs));
```

Comment:
We specifically want to catch network errors here. If `func()` throws for a non-network error (like perhaps the server responds with 'Unauthorized', or some other error) we shouldn't keep retrying